engine/profiler/src/profile_render.cpp

```diff
@@ -632,6 +635,7 @@ namespace dmProfileRender
             return;
         }
 
+        uint32_t last_thread_name = GetSelectedThread(render_profile, render_profile->m_CurrentFrame)->m_NameHash;
```

Comment:
Initial problem was in clearing `render_profile->m_CurrentFrame` almost every frame. When clear happened `last_thread->m_NameHash` become 0 that led to reset m_MaxFrameTime and reassign peak frame every time.